barber.h

```diff
@@ -1,27 +1,32 @@
 #ifndef BARBER_H
 #define BARBER_H
 
-#include <iostream>
 #include <thread>
-#include <chrono>
-
-using namespace std;
+#include <future>
 
 class Shop;
+class Semaphore;
 
 class Barber
 {
 public:
     Barber(Shop* shop);
-    void terminate();
+    void joinThread();
+    bool stopRequested();
```

Comment:
Could this be const? Could it be private?

Reply:
Declared as private :ok: